sockeye/inference.py

```diff
@@ -1085,7 +1063,8 @@ def _beam_search(self,
             lengths += mx.nd.cast(1 - mx.nd.expand_dims(finished, axis=1), dtype='float32')
 
             # (6) determine which hypotheses in the beam are now finished
-            finished = ((best_word_indices == C.PAD_ID) + (best_word_indices == self.vocab_target[C.EOS_SYMBOL]))
+            finished += ((best_word_indices == C.PAD_ID) + (best_word_indices == self.vocab_target[C.EOS_SYMBOL]))
+            finished = mx.nd.clip(finished, a_min=0, a_max=1)
```

Comment:
Is the clipping actually necessary? I think any positive value will evaluate to true.

Reply:
It is because we are using finished above to update lengths. So it should only contain values of 0 and 1

Reply:
we still guarantee that models can only produce the PAD symbol once finished. So why has it become necessary to do `+=` and `mx.nd.clip`?